### PR TITLE
[FEAT] Place Collectibles

### PR DESCRIPTION
**Pull request by Rofrtd**
# Description

This PR implements the placement of collectibles and resizes all the collectible images.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/island/collectibles/components/GoblinCrown.tsx
Comment (Rofrtd) [reply src/features/island/collectibles/components/GoblinCrown.tsx]: Eagle Eyes

**Comment by Polysemouse** [comment]
@Rofrtd  I think we can programatically set the width/scale of the image for the collectables / placables without manually setting them according to the pixel width of the .png or .gif

Consider the following placable:
``` tsx
<img
  style={{
    opacity: 0, // hide imgae before onLoad() is called
    left: `${GRID_WIDTH_PX * 35}px`,
    top: `${GRID_WIDTH_PX * 28}px`,
  }}
  id={Section.Fountain}
  onClick={() => {
    //Checks if fountainAudio is playing, if false, plays the sound
    if (!fountainAudio.playing()) {
      fountainAudio.play();
    }
  }}
  onLoad={setImageScale}
  className="absolute hover:img-highlight cursor-pointer"
  src={fountain}
  alt="Fountain"
/>
```

And a public method in some other file (eg. putting it in src/lib/image.ts)
``` tsx
export const setImageScale = (e: React.SyntheticEvent<HTMLImageElement, Event>): void => {
  if (e.currentTarget?.width) {
    e.currentTarget.style.setProperty("width", `${PIXEL_SCALE * e.currentTarget.width}px`);
    e.currentTarget.style.setProperty("opacity", "1");  // show image now the size is correct
  }
};
```

provided that the image pixel scale is already 1:1 in the .png or .gif file.

What do you think?

**Comment by adamhannigan** [comment]
@Polysemouse We had this in earlier versions but unfortunately it caused a quick flash and a bit of lag when a bunch of images were dynamically changing size

**Review by adamhannigan — DISMISSED** [review]
LGTM! 💪 🏆

**Comment by adamhannigan** [comment]
@Rofrtd Let's hold off on the Halloween event before merging this in. I predict there may be a few conflicts as they are changing some of the assets to add Halloween effects.

It may be easier when we revert the Halloween commit to merge this in

**Comment by Polysemouse** [comment]
> unfortunately it caused a quick flash and a bit of lag when a bunch of images were dynamically changing size

This happens to me too if the opacity of the images are not 0 before adjusting the sizes.  I guess it's okay to manually set the image sizes now untill we find a better solution

**Comment by spencerdezartsmith** [comment]
> 

I would love to see this too. I reckon we can play around with a solution like this down the track. 🤩

**Pull request merged**
